### ISSUE TITLE
Verify function rejects promise on successful operation

### ISSUE THREAD

**Issue by modev** [astra137/node-mojang#5]
When I was writing my the comments for my PR #4 I realized a bug which can get quite annoying. 
When calling _mojang.validate(access_token, client_token)_ with a valid access token/client token combination the promise gets rejected, since the operation returns an empty payload and if the data response is empty the promise gets rejected. 
However calling the function with a invalid access_token the promise gets resolved, since 
```javascript
if(data.length < 1){
  return reject(new Error("No data received."));
}
```
rejects the empty payload. 
I think this isn't too hard to fix, but probably it'd help if somebody would implement tests for this project first.

**Comment by RicoBrase** [comment]
Can confirm this, encountered this bug today as well.

Also throws "UnhandledPromiseRejectionWarning" and "DeprecationWarning".

    (node:14870) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: No data received.
    (node:14870) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

Note, that this crashes the NodeJS application (I'm running NodeJS 7.10.0 and the mojang-package is version 0.4.1).

**Comment by astra137** [comment]
Should >=400 error codes cause the promise to reject? I want to say that it is more normal a behavior, but I have no evidence and a load of bias. It would also be a breaking change.

**Comment by modev** [comment]
Yes, it would be a breaking change, but it would represent the API in a better, more logical way. The problem about the code-snipped is, that some functions API endpoints, like verify, return an empty payload if **successful**. This causes the promise to reject if the token is valid and to resolve if the token is invalid. 
I think this is not a good implementation at all, but its part of Mojangs *awful* API design.

**Comment by astra137** [comment]
I would recommend looking at the status code and parsing content only if it is available. I am using `got` in my fork now while I change the methods in that manner, but the _request file could also be easily enough be upgraded to handle server responses that way too. I tend to stay away from using http/https directly, but I guess it couldn't hurt. Plus zero-dependency libraries are nice. So either way.

**Comment by dscalzi** [comment]
Should be resolved by #13
